api/fasit.go

```diff
@@ -119,6 +200,85 @@ func (fasit FasitClient) getResource(resourcesRequest ResourceRequest, environme
 	return resource, nil
 }
 
+func (fasit FasitClient) createResource(resource ExposedResource, environment, application, zone string) (int, error) {
+	requestCounter.With(nil).Inc()
+
+	payload := bytes.NewBuffer([]byte(buildResourcePayload(resource, environment, application, zone)))
+	req, err := http.NewRequest("POST", fasit.FasitUrl+"/api/v2/resources/", payload)
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to create request: %s", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+
+	httpReqsCounter.WithLabelValues(strconv.Itoa(resp.StatusCode), "POST").Inc()
+	if resp.StatusCode > 299 {
+		errorCounter.WithLabelValues("error_fasit").Inc()
+		return 0, fmt.Errorf("Fasit returned: %s (%s)", body, strconv.Itoa(resp.StatusCode))
+	}
+
+	type CreatedResource struct {
+		Id int
+	}
+	createdResource := new(CreatedResource)
+	if err := json.Unmarshal(body, &createdResource); err != nil {
+		errorCounter.WithLabelValues("read_body").Inc()
+		return 0, fmt.Errorf("Could not read response: %s", err)
+	}
+
+	return createdResource.Id, nil
+}
+
+func (fasit FasitClient) getFasitEnvironment(environmentName string) (error) {
+	requestCounter.With(nil).Inc()
+	req, err := http.NewRequest("GET", fasit.FasitUrl+"/api/v2/environments/"+environmentName, nil)
+	if err != nil {
+		return fmt.Errorf("Could not create request: %s", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	defer resp.Body.Close()
+
+	if err != nil {
+		return fmt.Errorf("Error contacting Fasit: %s", err)
+	}
+
+	if resp.StatusCode == 200 {
+		return nil
+	}
+	return fmt.Errorf("Could not find environment %s in Fasit", environmentName)
+}
+
+func (fasit FasitClient) getFasitApplication(application string) (error) {
```

Comment:
- no need for parens with single value return. 
- can use custom error struct with http status here to make it possible for caller to know if it was 404 (in which case a resource should be created)